src/js/controller/tracks-loader.js

```diff
@@ -0,0 +1,110 @@
+define([
+    'utils/underscore',
+    'utils/helpers',
+    'parsers/parsers',
+    'parsers/captions/srt',
+    'parsers/captions/dfxp',
+    'parsers/captions/vttcue'
+], function(_, utils, parsers, srt, dfxp, VTTCue) {
+    var tracksLoader = {};
+
+    tracksLoader.loadFile = function(track, successHandler, errorHandler) {
+        track.xhr = utils.ajax(track.file, function(xhr) {
+            xhrSuccess(xhr, track, successHandler, errorHandler);
```

Comment:
This is not called in the right context:

`xhrSuccess.call(tracksLoader, xhr, track, successHandler, errorHandler);`

See squash/bootstrap.html?primary=html5&edition=ads&feature=api_set_captions scenario 2